modules/browsiBidAdapter.js

```diff
@@ -0,0 +1,171 @@
+          schain: schain,
+          params: params
+        }
+      };
+      requests.push(request);
+    })
+    return requests;
+  },
+  /**
+   * Unpack the response from the server into a list of bids.
+   * @param serverResponse A successful response from the server.
+   * @param request
+   * @returns {Bid[]} An array of bids which were nested inside the server.
+   */
+  interpretResponse: function (serverResponse, request) {
+    const bidResponses = [];
+    const response = serverResponse?.body;
+    if (!response) { return bidResponses; }
+    const {
+      bidId,
+      w,
+      h,
+      vXml,
+      vUrl,
+      cpm,
+      cur,
+      ttl,
+      ...extraParams
+    } = response;
+    delete extraParams.userSyncs;
+    const bidResponse = {
+      requestId: request.data.bidId,
+      bidId,
+      vastXml: vXml,
+      vastUrl: vUrl,
+      cpm,
+      ttl,
+      mediaType: VIDEO,
+      width: w,
+      height: h,
+      currency: cur,
+      bidderCode: BIDDER_CODE,
+      ...extraParams
+    };
+    bidResponses.push(bidResponse);
+    return bidResponses;
+  },
+  /**
+   * Extracts user-syncs information from server response
+   * @param syncOptions {SyncOptions}
+   * @param serverResponses {ServerResponse[]}
+   * @param gdprConsent
+   * @param uspConsent
+   * @returns {UserSync[]}
+   */
+  getUserSyncs: function (syncOptions, serverResponses, gdprConsent, uspConsent) {
+    const serverResponse = isArray(serverResponses) ? serverResponses[0] : serverResponses;
+    const syncParams = serverResponse?.body?.userSyncs;
+    const userSyncs = [];
+    const allowedTypes = [];
+    syncOptions.iframeEnabled && allowedTypes.push('iframe');
+    syncOptions.pixelEnabled && allowedTypes.push('image');
+    if (syncParams && allowedTypes.length) {
+      syncParams.forEach(syncParam => {
+        let { url, type } = syncParam;
+        if (!allowedTypes.includes(type)) { return; }
+        url = getValidUrl(url, gdprConsent, uspConsent);
+        userSyncs.push({
+          type,
+          url
+        });
+      })
+    }
+    return userSyncs;
+  },
+  onTimeout(timeoutData) {
+    logInfo(`${BIDDER_CODE} bidder timed out`, timeoutData);
+  },
+  onBidderError: function ({error}) {
+    logError(`${BIDDER_CODE} bidder error`, error);
+  }
+}
+/**
+ * Replaces GdprConsent and uspConsent params in url
+ * @param url {String}
+ * @param gdprConsent
+ * @param uspConsent
+ * @returns {string}
+ */
+const getValidUrl = function (url, gdprConsent, uspConsent) {
+  let validUrl = url.replace(GDPR_TO_REPLACE, gdprConsent?.gdprApplies || '')
+    .replace(GDPR_STR_TO_REPLACE, encodeURIComponent(gdprConsent?.consentString || ''))
+    .replace(USP_TO_REPLACE, encodeURIComponent(uspConsent?.consentString || ''));
+  if (validUrl.indexOf('http') < 0) {
+    validUrl = 'http://' + validUrl;
+  }
+  return validUrl;
+}
+
+const validateBrowsiIds = function (pubId, tagId) {
+  return isStr(pubId) && isStr(tagId);
```

Comment:
`pubId` & `tagId` are required but they can also be empty strings. Is the expected?

Reply:
its not expected - fixed